packages/lib/onedrive-api.ts

```diff
@@ -339,6 +341,15 @@ export default class OneDriveApi {
 
 					await handleRequestRepeat(error);
 					continue;
+				} else if (error?.code === 'activityLimitReached' && response?.headers?._headers['retry-after'][0]) {
+					// Wait for OneDrive throttling
+					// Relavent Microsoft Docs: https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#best-practices-to-handle-throttling
+					// Decrement retry count as multiple sync threads will cause repeated throttling errors - this will wait until throttling is resolved to continue, preventing a hard stop on the sync
+					i--;
+					const sleep_seconds = response.headers._headers['retry-after'][0];
+					logger.info(`OneDrive Throttle, sync thread sleeping for ${sleep_seconds} seconds...`);
+					await handleRequestRepeat(error, parseInt(sleep_seconds));
```

Comment:
parseInt is kind of broken because if you don't provide a radix it will try to guess and sometimes does it incorrectly. I think it's best to stick to `Number(string)`